code/datums/components/food/decomposition.dm

```diff
@@ -0,0 +1,162 @@
+	var/obj/item/food/food = parent
+	var/atom/last_loc = food.loc
+
+	var/clean = FALSE // Used to check if it's on a clean surface
+	while(last_loc && !isarea(last_loc))
+		if(HAS_TRAIT(food.loc, TRAIT_PROTECT_FOOD))
+			clean = TRUE
+			break
+		else if(locate(/obj/structure/table) in food.loc)
+			clean = TRUE
+			break
+		last_loc = last_loc.loc
+	if(!handled || clean || protected)
+		// prevent decomposition
+		if(active_timers)
+			time_remaining = timeleft(timerid)
+			deltimer(timerid)
+		return
+	// do decomposition
+	timerid = addtimer(CALLBACK(src, .proc/decompose), time_remaining, TIMER_STOPPABLE | TIMER_UNIQUE) //Was told to use this instead of processing I guess
+
+/datum/component/decomposition/Destroy()
+	if(active_timers)
+		deltimer(timerid) //Just in case
+	return ..()
+
+/datum/component/decomposition/proc/dropped()
+	SIGNAL_HANDLER
+	protected = FALSE
+	handle_movement()
+
+/datum/component/decomposition/proc/picked_up()
+	SIGNAL_HANDLER
+	if(active_timers)
+		time_remaining = timeleft(timerid)
+		deltimer(timerid)
+	protected = TRUE
+	if(!handled)
+		handled = TRUE
+
+/datum/component/decomposition/proc/decompose()
+	var/obj/item/food/decomp = parent //Lets us spawn things at decomp
+	new /obj/effect/decal/cleanable/ants(decomp.loc)
+	new /obj/item/food/badrecipe/moldy(decomp.loc)
+	decomp.visible_message("<span class='notice'>[decomp] gets overtaken by mold and ants! Gross!</span>")
+	qdel(decomp)
+	return
+
+/datum/component/decomposition/proc/examine(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+	if(active_timers) // Is the timer currently applied to this?
+		switch(examine_type)
+			if(0)// All other types
+				switch(timeleft(timerid)) // Deciseconds used so there's no gaps between examine times.
+					if(3001 to 4500) // 7.5 to 5 Minutes left
+						examine_list += "[parent] looks kinda stale."
+					if(1501 to 3000) // 5 to 2.5 Minutes left
+						examine_list += "[parent] is starting to look pretty gross."
+					if(1 to 1500) // 2.5 Minutes to 1 Decisecond left
+						examine_list += "[parent] looks barely edible."
+			if(1) // Gross food
+				switch(timeleft(timerid))
+					if(2101 to 3150) // 5.25 to 3.5 Minutes
+						examine_list += "[parent] looks kinda stale."
+					if(1050 to 2100) // 3.5 to 1.75 Minutes left
+						examine_list += "[parent] is starting to look pretty gross."
+					if(1 to 1051) // 1.75 Minutes to 1 Decisecond left
+						examine_list += "[parent] looks barely edible."
+			if(2) // Raw food
+				switch(timeleft(timerid))
+					if(1501 to 2250) // 3.75 to 2.5 Minutes left
+						examine_list += "[parent] looks kinda stale."
+					if(751 to 1500) // 2.5 to 1.25 Minutes left
+						examine_list += "[parent] is starting to look pretty gross."
+					if(1 to 750) // 1.25 Minutes to 1 Decisecond left
+						examine_list += "[parent] looks barely edible."
+	else // No timer currently running.
+		switch(examine_type)
```

Comment:
What is the value of examine_type? is it a define? you're looking for these arbitrary values, like 0, 1, 2, what do they mean?

Reply:
They're for the RAW, GROSS, and normal timers. I should have defined those, I'll fix that